src/testing.rs

```diff
@@ -216,7 +219,8 @@ impl crate::voter::Environment<&'static str, u32> for Environment {
 		_round: u64,
 		_state: RoundState<&'static str, u32>,
 		_base: (&'static str, u32),
-		_votes: &HistoricalVotes<&'static str, u32, Self::Signature, Self::Id>,
```

Comment:
Isn't the `_votes` parameter redundant now? Everything we've observed should be in `HistoricalVotes`.